runtime/queries/sourcepawn/highlights.scm

```diff
@@ -0,0 +1,326 @@
+
+(condition_statement) @conditional
+
+(while_statement) @repeat
+
+(do_while_statement) @repeat
+
+(switch_statement) @conditional
+
+(switch_case) @conditional
+
+(ternary_expression) @conditional.ternary
+
+; Expressions
+(view_as) @function.builtin
+
+(sizeof_expression) @function.macro
+
+(this) @variable.builtin
+
+; https://github.com/alliedmodders/sourcemod/blob/5c0ae11a4619e9cba93478683c7737253ea93ba6/plugins/include/handles.inc#L78
+(hardcoded_symbol) @variable.builtin
+
+; Comments
+(comment) @comment
+
+; General
+(parameter_declaration
+  defaultValue: (identifier) @constant)
+
+(fixed_dimension) @punctuation.bracket
+
+(dimension) @punctuation.bracket
+
+(array_indexed_access) @punctuation.bracket
+
+(escape_sequence) @string.escape
+
+; Constructors
+(new_expression
+  class: (identifier) @type
+  arguments: (call_arguments) @constructor)
+
+; Methodmaps
+(methodmap) @type.definition
+
+(methodmap
+  name: (identifier) @type)
+
+(methodmap
+  inherits: (identifier) @type)
+
+(methodmap_method_constructor
+  name: (identifier) @constructor)
+
+(methodmap_method
+  name: (identifier) @method)
+
+(methodmap_native
+  name: (identifier) @method)
+
+(methodmap_property
+  name: (identifier) @property)
+
+(methodmap_property_getter) @method
+
+(methodmap_property_setter) @method
+
+; Enum structs
+(enum_struct) @type.definition
+
+(enum_struct
+  name: (identifier) @type)
+
+(enum_struct_field
+  name: (identifier) @field)
+
+(enum_struct_method
+  name: (identifier) @method)
+
+; Non-type Keywords
+(variable_storage_class) @storageclass
+
+(visibility) @storageclass
+
+(visibility) @storageclass
+
+(assertion) @function.builtin
+
+(function_declaration_kind) @keyword.function
+
+[
+  "new"
+  "delete"
+] @keyword.operator
+
+[
+  "."
+  ","
+] @punctuation.delimiter
+
+; Operators
+[
+  "+"
+  "-"
+  "*"
+  "/"
+  "%"
+  "++"
+  "--"
+  "="
+  "+="
+  "-="
+  "*="
+  "/="
+  "=="
+  "!="
+  "<"
+  ">"
+  ">="
+  "<="
+  "!"
+  "&&"
+  "||"
+  "&"
+  "|"
+  "~"
+  "^"
+  "<<"
+  ">>"
+  ">>>"
+  "|="
+  "&="
+  "^="
+  "~="
+  "<<="
+  ">>="
+] @operator
+
+(ignore_argument) @operator
+
+(scope_access) @operator
+
+(rest_operator) @operator ; Should override (concatenated_string) but currently does nothing
+
+; public Plugin myinfo
+(struct_declaration
+  name: (identifier) @variable.builtin)
+
+; Typedef/Typedef
+(typeset) @type.definition
+
+(typedef) @type.definition
+
+(functag) @type.definition
+
+(funcenum) @type.definition
+
+(typedef_expression) @keyword.function ; function void(int x)
+
+; Enums
+(enum) @type.definition
+
+(enum
+  name: (identifier) @type)
+
+(enum_entry
+  name: (identifier) @constant)
+
+(enum_entry
+  value: (_) @constant)
+
+; Literals
+(int_literal) @number
+
+(char_literal) @character
+
+(float_literal) @float
+
+(string_literal) @string
+
+(array_literal) @punctuation.bracket
+
+(concatenated_string) @punctuation.delimiter ; Dots in string-literal concat--Doesn't seem to work?
+
```

Comment:
If it doesn't work, it should either be fixed or removed.

```suggestion
```

Reply:
Leftover from a previous commit, it works 👍